xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-notifiers/xwiki-platform-notifications-notifiers-default/src/main/java/org/xwiki/notifications/notifiers/internal/UserEventDispatcher.java

```diff
@@ -116,165 +114,178 @@ public class UserEventDispatcher implements Runnable, Disposable, Initializable
     private EventStore events;
 
     @Inject
-    private Logger logger;
+    private RecordableEventDescriptorManager recordableEventDescriptorManager;
 
-    @Override
-    public void initialize() throws InitializationException
-    {
-        // Live events (event which just been produced) are put in the priority queue to be dealt with faster and we put
-        // events coming from some migration or restart in a larger queue but containing only the id to not flood the
-        // live events queue. If the priority queue is full (very very active wiki or some mistake in a script) events
-        // will fallback on the large queue. If both queue are full we just ignore the event and it will be picked up by
-        // the next missing pre filtered event run.
-        this.priorityQueue = new LinkedBlockingQueue<>(1000);
-        this.secondaryQueue = new LinkedBlockingQueue<>(100000);
-
-        // Start a background thread to filter and dispatch users events
-        Thread thread = new Thread(this);
-        thread.setName("User event dispatcher thread");
-        thread.setPriority(Thread.NORM_PRIORITY - 1);
-        thread.setDaemon(true);
-        thread.start();
-    }
+    @Inject
+    private RemoteObservationManagerConfiguration remoteObservation;
 
-    @Override
-    public void dispose() throws ComponentLifecycleException
-    {
-        // Mark the component as disposed
-        this.disposed = true;
+    @Inject
+    private ExecutionContextManager contextManager;
 
-        // Make sure to wake the thread up to stop it
-        wakeup();
-    }
+    @Inject
+    private Execution execution;
 
-    private void wakeup()
+    @Inject
+    private Logger logger;
+
+    private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+
+    private volatile boolean running;
+
+    /**
+     * Start the scheduler.
+     */
+    public void initialize()
```

Comment:
To be honest I find a bit weird this initialize method while the component is not Initializable anymore. Might be useful to provide a bit more explanation in javadoc.

Reply:
I really don't see what's weird about a component with an #initialize() method, and you have a javadoc explaining what it's about.

The point of Initializable is to automate the initialization as soon as the component is initialized, but we don't want that here, we just initialize it when we are ready.

Reply:
It's just that we often have `initialize` method for classes implementing `Initializable` so it's rare to perform an explicit call to it. So that's why I was suggesting to be more explicit about that here. But no big deal especially since it's not something expected to be reused in many places.